docs/dev/analysis/features/numbering.rst

```diff
@@ -2,54 +2,703 @@
+Full API support would imply access to a "Level 5"
+representation.
+Decisions about simplifying the implementation of
+numbered lists w/r/t the ISO standard are most
+likely to come up when developing "Level 5"
+support.
+However, it should be uncontroversial to adhere
+to the ISO standard very closely when developing
+"Level 4" support. This means that the level of
+*support* for advanced list operations can
+be increased through some straightforward 
+**xmlchemy**-based declarations,
+and high-level design questions need not come
+into play.
+
+This is done through two main programming tasks:
+
+1.  Declare relevant XML types following the ``wml.xsd`` schema
+    (ISO-29500-1 Appendix A.1).
+2.  Instruct ``xmlchemy`` to recognize a relevant tag 
+    as an instance of the appropriate type.
+
+If this is accomplished, various low-level methods will
+be exposed which abstract the necessary XML manipulations,
+allowing for improved access to desired functions for
+a user familiar with the semantics of the ISO standard.
+
+Pull Request #XX is concerned with updating the module
+to declare the following types and expose them to
+``xmlchemy``-based methods:
+
++-------------------+
+|XML Type           |
++===================+
+|CT_NumPicBullet    |          
++-------------------+
+|CT_AbstractNum     |         
++-------------------+
+|CT_LongHexNumber   |           
++-------------------+
+|ST_LongHexNumber   |           
++-------------------+
+|CT_MultiLevelType  |            
++-------------------+
+|ST_MultiLevelType  |            
++-------------------+
+|CT_Lvl             | 
++-------------------+
+|CT_NumFmt          |    
++-------------------+
+|ST_NumberFormat    |          
++-------------------+
+|CT_LevelSuffix     |         
++-------------------+
+|ST_LevelSuffix     |         
++-------------------+
+|CT_LevelText       |       
++-------------------+
+|CT_LvlLegacy       |       
++-------------------+
+|CT_Num             | 
++-------------------+
+
+Having defined these types and trained the parser to
+associate them with elements in the namespace
+(this is done through calls to ``register_element_cls``
+in ``oxml.__init__``), it is possible to implement
+solutions to the documented issues noted above in
+a disciplined way.
+
+Making use of low-level support
+-------------------------------
+Once the types listed above are defined and the **xmlchemy**
+submodule methods can be used, it becomes a little less
+painful to implement a solution to the StackOverflow
+question referred to above.
+
+
+.. code-block:: python
+
+
+    #!/usr/bin/python
+    
+    from docx import Document
+    from docx import oxml
+    
+    
+    d = Document()
+    
+    
+    """
+    1. Create an abstract numbering definition for a multi-level numbering style.
+    """
+    numXML = d.part.numbering_part.numbering_definitions._numbering
+    nextAbstractId = max([ J.abstractNumId for J in numXML.abstractNum_lst ] ) + 1
```

Comment:
@jlovegren0 I'm using your fork installation (from source) and running the script thereon:

some observations: https://github.com/python-openxml/python-docx/blob/5bd5cf6a54a2a5585c05f79279da63ce78b8e828/docx/oxml/numbering.py#L253

this does not have a attr **abstractNum_lst** and the script complains with the following error:

```
nextAbstractId = max([J.abstractNumId for J in numXML.abstractNum_lst]) + 1
AttributeError: 'CT_Numbering' object has no attribute 'abstractNum_lst'
```

so I changed it to expect the declared attr **abstractNum** and still get the error:
```
nextAbstractId = max([J.abstractNumId for J in numXML.abstractNum]) + 1
AttributeError: 'CT_Numbering' object has no attribute 'abstractNum'
```

looking further, found that **numXML** object has a list object **num_lst** whose each element (CT_ Num type) has **abstractNumId**. (see attachment) so using this instead the script fails with:

```
nextAbstractId = max([J.abstractNumId for J in numXML.num_lst]) + 1
TypeError: '>' not supported between instances of 'CT_DecimalNumber' and 'CT_DecimalNumber'
```

I was looking at the python-docx source code to see what makes sense and surprising part is that my IDE shows **abstractNum_lst** as a suggestion in the drop down when hitting up on numXML attrs. (see attachment)

Can you please help identify if I'm missing something obvious or if there's a discrepency somewhere in the usage? (I'm on python37, not sure if that matters but for now looks like I need to check every bit)

Reply:
<img width="1041" alt="Screenshot 2020-04-01 at 3 35 52 PM" src="https://user-images.githubusercontent.com/1414265/78127251-726a2580-7431-11ea-9b63-eb613f415fa0.png">
<img width="523" alt="Screenshot 2020-04-01 at 3 40 09 PM" src="https://user-images.githubusercontent.com/1414265/78127270-78600680-7431-11ea-86fc-1711df529e47.png">

Reply:
please ignore this issue for this PR for now. I am able to use your script using local modules.